Modules/System/Cryptography Management/src/Xml/EncryptedXml.Codeunit.al

```diff
@@ -0,0 +1,68 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+/// <summary>
+/// Provides functionality for encrypting and decrypting XML documents.
+/// </summary>
+codeunit 1465 EncryptedXml
+{
+    Access = Public;
+
+    var
+        EncryptedXmlImpl: Codeunit "EncryptedXml Impl.";
+
+    /// <summary>
+    /// Creates a symmetric session key to encrypt a XML document and then 
+    /// uses the X.509 certificate to embed an encrypted version of the session key in the XML document.
+    /// </summary>
+    /// <param name="XmlDocument">The XmlDocument to encrypt.</param>
+    /// <param name="ElementToEncrypt">The name of the element to encrypt.</param>
+    /// <param name="X509CertBase64Value">The X509Certificate2 to use for the asymmetric encryption.</param>
+    [NonDebuggable]
+    procedure Encrypt(var XmlDocument: XmlDocument; ElementToEncrypt: Text; X509CertBase64Value: Text)
```

Comment:
Can you please add support for certs with passwords ? We have most of the cases where certs have pass.

Reply:
Yes, of course, that is better. Fixed now.